svm/src/program_loader.rs

```diff
@@ -69,8 +69,8 @@ pub(crate) fn load_program_accounts<CB: TransactionProcessingCallback>(
     callbacks: &CB,
     pubkey: &Pubkey,
 ) -> Option<ProgramAccountLoadResult> {
-    let program_account = callbacks.get_account_shared_data(pubkey)?;
-
+    // load program account by `pubkey` and don't insert the program account into read cache.
+    let (program_account, _slot) = callbacks.load_account_with(pubkey, |_| false)?;
```

Comment:
@Lichtso 
Here we don't populate readcache for missing program loading.

Reply:
But we do put programdata_account into read cache on line 106. 
Maybe we should skip puting programdata_account from read cache too?